### PR TITLE
Clarifies site init param for DD possible org locations

### PR DESCRIPTION
**Pull request by tbavelier**
## Motivation

Adds clarification upon the `site` init parameter for log collection based on a recent support request

## Changes

* Changes the table with the init parameters to add more entries to the `site` option, as well as a link towards the doc to understand the site
* Unification with RUM doc
## Testing

Check the `.md` appearance following changes

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Review by BenoitZugmeyer — COMMENTED** [review]
Thank you for your contribution.

I think we should unify this documentation with the [RUM documentation](https://github.com/DataDog/browser-sdk/blob/main/packages/rum/README.md#initialization-parameters). Specifically:

* don't list everything here, a link to the documentation is enough
* the link you are using looks great, much better than the one in the RUM documentation. Could you replace it there too?

**Comment by tbavelier** [comment]
Hey Benoit,

Added the changes you mentioned. Examples might be useful in my opinion, as for US3 the value would be `us3.datadoghq.com` (same value as in my link) but for US1, the value would be `datadoghq.com` (i.e. without the `app.`)

File: packages/rum/README.md [hunk 1]
```diff
@@ -361,7 +361,7 @@ A service is an independent, deployable code repository that maps to a set of pa
 [11]: https://www.datadoghq.com/pricing/?product=real-user-monitoring--session-replay#real-user-monitoring--session-replay
 [12]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces?tab=browserrum
 [13]: https://docs.datadoghq.com/real_user_monitoring/session_replay/privacy_options?tab=maskuserinput
-[14]: https://docs.datadoghq.com/agent/basic_agent_usage#datadog-site
+[14]: https://docs.datadoghq.com/getting_started/site/
```
Comment (bcaudan) [inline packages/rum/README.md:364]: The new URL is nicer but it does not specify the value of the param to put in the configuration though 😕
Should we update this doc to list which parameter to use for each DC?
something like:
```
SITE | SITE URL | SITE PARAMETER | LOCATION
US1 | https://app.datadoghq.com | datadoghq.com | US
US3 | https://us3.datadoghq.com | us3.datadoghq.com | US
```

Comment (tbavelier) [reply packages/rum/README.md:364]: Doc PR was merged (not deployed yet) and updated this PR accordingly

**Review by alai97 — APPROVED** [review]

**Review by bcaudan — APPROVED** [review]
Nice 👍

**Review by BenoitZugmeyer — APPROVED** [review]

**Pull request merged**